### PR TITLE
test: added PowerMocked unit tests for the ConnectivityUtils class.

### PR DESCRIPTION
**Pull request by ardevd**
Added necessary PowerMock dependencies in addition to a JSON dependency which is required in order to unit test with JSON objects. The default JSON library is part of AOSP and would need to be mocked which is not what we want.

Note: the test currently fails. The `isInternetWalled` method has a weird check that I do not quite understand:

https://github.com/nextcloud/android/blob/b252c6440d196409076b7a00c83ef4271cbae1ef/src/main/java/com/owncloud/android/utils/ConnectivityUtils.java#L76-L89

So only if we're on NextCloud v13 we do the `else` block?

My test fails if I set version to `OwnCloudVersion.nextcloud_14` and true if I set it to `OwnCloudVersion.nextcloud_13`.

Could someone shed some light on the intended functionality here? I'm confused.

### CODE REVIEW

**Comment by ardevd** [comment]
@tobiasKaminsky Nvm, I'm fairly sure `if (serverVersion.compareTo(OwnCloudVersion.nextcloud_13) > 0)` will only be true for NextCloud versions > 13. With this assumption I'll update the tests and this PR.

**Comment by ardevd** [comment]
this signoff requirement is driving me nuts. can someone fix this? My git fu is not strong enough it seems.

**Comment by tobiasKaminsky** [comment]
> @tobiasKaminsky Thanks for the explanation. But what about version > NC13?

Sorry, that was a typo on my side, it is >= NC13

**Comment by tobiasKaminsky** [comment]
> Sorry, that was a typo on my side, it is >= NC13

Mockup seems to have problem when using ">=" in a list. I added ">=" above, but only "=" was shown.

File: src/test/java/com/owncloud/android/utils/ConnectivityUtilsTest.java [hunk 1]
```diff
@@ -45,11 +64,27 @@
     @Mock
     private NetworkInfo mNetworkInfo;
 
+    @Mock
+    private Account mAccount;
```
Comment (tobiasKaminsky) [inline src/test/java/com/owncloud/android/utils/ConnectivityUtilsTest.java]: We try to get rid of the m prefix everywhere.
Can you remove them here? So we do not introduce now ones :-)

**Comment by tobiasKaminsky** [comment]
:+1: nice, so now we have a blue print how to mock static methods.

Thank you very much for your contribution!

File: src/test/java/com/owncloud/android/utils/ConnectivityUtilsTest.java (hunk 1 above)
Comment (ardevd) [reply src/test/java/com/owncloud/android/utils/ConnectivityUtilsTest.java]: done. :)

File: src/test/java/com/owncloud/android/utils/BitmapUtilsTest.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import java.net.URL;
+
+import static org.junit.Assert.*;
+
+public class BitmapUtilsTest {
+
+
+    @Test
+    public void calculateHSL_assertCorrectRGBArrayReturned() throws Exception {
+        int[] rgbArray = BitmapUtils.calculateHSL("Edvard");
```
Comment (tobiasKaminsky) [inline src/test/java/com/owncloud/android/utils/BitmapUtilsTest.java]: This should be a real test with assert?

File: src/test/java/com/owncloud/android/utils/BitmapUtilsTest.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    }
+
+    @Test
+    public void HSLtoRGB_assertCorrectRGBArrayReturned() throws Exception {
+        int[] rgbArray = BitmapUtils.HSLtoRGB(300.0f, 90.0f, 99.0f, 1.0f);
+        String format = "hello";
```
Comment (tobiasKaminsky) [inline src/test/java/com/owncloud/android/utils/BitmapUtilsTest.java]: Same here

Comment (ardevd) [reply src/test/java/com/owncloud/android/utils/BitmapUtilsTest.java]: oopsies. that class should not have been committed. Removed.

File: src/test/java/com/owncloud/android/utils/BitmapUtilsTest.java (hunk 2 above)
Comment (ardevd) [reply src/test/java/com/owncloud/android/utils/BitmapUtilsTest.java]: my bad.

**Pull request merged**